src/main/java/edu/isi/gaia/ColdStart2AidaInterchange.kt

```diff
@@ -204,9 +202,14 @@ class ColdStart2AidaInterchangeConverter(
 //                associate_with_system(realisNode)
 //            }
 
+            val justificationType = provenanceToMentionType[cs_assertion.justifications]
+                    .asSequence()
+                    .maxWith(Ordering.explicit(listOf(CANONICAL_MENTION, PRONOMIAL_MENTION,
+                            NOMINAL_MENTION, NORMALIZED_MENTION, NAME_MENTION)))!!
+
             registerJustifications(entityResource, cs_assertion.justifications,
                     typeAssertion, cs_assertion.string, confidence,
-                    justificationType = cs_assertion.mentionType.name)
+                    justificationType = justificationType.name)
```

Comment:
Was the issue that the kind of mention in the `justificationType` was getting dropped?

Reply:
Yes - in ColdStart the single "best" mention for an entity will have two entries - one with its actual mention type and one with `canonical_mention`.  The bug was that `canonical_mention` was clobbering the true mention types of the canonical mentions.